modules/internal/DefaultAssociative.chpl

```diff
@@ -944,6 +935,35 @@ module DefaultAssociative {
     }
   }
 
+  proc chpl_serialReadWriteAssociativeHelper(f, arr, dom) throws
+  where _usingSerializers(f) && !_isDefaultDeser(f) {
+      ref fmt = if f._writing then f.serializer else f.deserializer;
+
+      if f._writing then
+        fmt.startMap(f, dom.dsiNumIndices:uint);
+      else
+        fmt.startMap(f);
+
+      if f._writing {
+        for key in dom {
+          fmt.writeKey(f, key);
+          fmt.writeValue(f, arr.dsiAccess(key));
+        }
+      } else {
+        for 0..<dom.dsiNumIndices {
+          const k = fmt.readKey(f, dom.idxType);
+
+          if !dom.dsiMember(k) {
+            // TODO: throw error
```

Comment:
Did you mean to address this in this PR?

Reply:
Not at this time. It's not clear to me what kind of error should be thrown. I can update the comment to reflect that uncertainty.